modules/javafx.graphics/src/main/java/javafx/stage/Stage.java

```diff
@@ -1194,6 +1194,10 @@ private void doVisibleChanged(boolean value) {
             if (peer != null) {
                 peer.setIcons(platformImages);
             }
+
+            if (!isIconified()) {
+                requestFocus();
```

Comment:
As for Android, and as discussed earlier in comments, this delays a little bit the focus request from the original implementation before JDK-8296621, but it works fine (the stage gets focused), in line with the earlier proposed fix in MonocleWindow:

```
04-24 14:06:45.970 31436 31461 D GraalCompiled: (Original focus request, before JDK-8296621, in WindowStage::setScene)
04-24 14:06:46.395 31436 31461 D GraalCompiled: (Earlier proposed focus request in MonocleWindow::setVisible)
04-24 14:06:46.395 31436 31461 D GraalCompiled:  (Current focus request in Stage::doVisibleChanged)
```